main/php.h

```diff
@@ -434,4 +434,10 @@ END_EXTERN_C()
 
 #include "php_reentrancy.h"
 
+/* the following typedefs are deprecated; use the standard C99 types
+ * instead */
+typedef bool zend_bool;
```

Comment:
if used, they should emit a compiler warning, also I would add something like "and will be removed in PHP 9.0"

Reply:
GCC should warn implicitly when `-Wno-deprecated-declarations` is not passed

php-src CI should not pass if any GCC warning is produced

Reply:
Yes, but I thought you meant that GCC should write "will be removed in 9" in the message. In C++, the `deprecated` attribute can be annotated with a message that is printed (https://en.cppreference.com/w/cpp/language/attributes/deprecated), but not the proprietary GCC attribute.
Will add the 9.0 comment.